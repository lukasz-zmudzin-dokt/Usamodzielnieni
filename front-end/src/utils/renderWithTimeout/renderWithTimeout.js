import React from "react";

export const renderWithTimeout = (Component, timeout_ms) => {
  return <Timeout timeout_ms={timeout_ms} children={Component} />;
};

class Timeout extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      visible: true,
    };
    this.interval = null;
  }

  componentDidMount() {
    this.interval = setTimeout(
      () => this.setState({ visible: false }),
      this.props.timeout_ms
    );
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  render() {
    const Child = this.props.children;
    return this.state.visible ? <Child /> : null;
  }
}
