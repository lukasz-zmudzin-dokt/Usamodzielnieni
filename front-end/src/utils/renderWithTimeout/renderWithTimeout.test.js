import React from "react";
import {render} from "@testing-library/react";
import {renderWithTimeout} from "./renderWithTimeout";

describe('renderWithTimeout', () => {
    let component, timeout;

    beforeEach(() => {
        component = jest.fn().mockImplementation(() => {
            return <div>środek komponentu</div>
        });
        timeout = 10000;
        jest.useFakeTimers();
        jest.clearAllMocks();
    });

    it('should render component', () => {
        const {container} = render(
            renderWithTimeout(component, timeout)
        );

        expect(container).toMatchSnapshot();
    });

    it('should hide component after certain time', () => {
        timeout = 10000;
        const {getByText, queryByText} = render(
            renderWithTimeout(component, timeout)
        );

        expect(setTimeout).toHaveBeenCalledTimes(1);
        expect(getByText("środek komponentu")).toBeInTheDocument();

        jest.runAllTimers();

        expect(queryByText("środek komponentu")).not.toBeInTheDocument();
    });
});