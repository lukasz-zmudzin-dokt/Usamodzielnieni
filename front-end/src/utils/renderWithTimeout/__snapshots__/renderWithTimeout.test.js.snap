// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderWithTimeout should render component 1`] = `
<div>
  <div>
    Å›rodek komponentu
  </div>
</div>
`;
